src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAdminController.java

```diff
@@ -142,10 +153,54 @@ public void fullDeleteStudy(Long[] ids) {
             UserSearch userSearch)
             throws UserNotFoundException {
 
-        Page<IUser> users = userService.getAll(pageable, userSearch);
+        Page<U> users = userService.getAll(pageable, userSearch);
         return users.map(user -> conversionService.convert(user, CommonUserDTO.class));
     }
 
+    @ApiOperation("Register new users")
+    @RequestMapping(value = "/api/v1/admin/users/batch", method = RequestMethod.POST)
+    public void register(@RequestBody BulkUsersRegistrationDTO bulkUsersDto) throws PasswordValidationException, ValidationException {
+
+        if (bulkUsersDto.getTenantIds() == null || bulkUsersDto.getTenantIds().isEmpty()) {
+            throw new ValidationException("tenants: must be not empty");
+        }
+
+        boolean emailConfirmationRequired = bulkUsersDto.getEmailConfirmationRequired();
+
+        Set<Tenant> tenants = bulkUsersDto.getTenantIds().stream()
+                .map(tenantId -> conversionService.convert(tenantId, Tenant.class))
+                .collect(Collectors.toSet());
+        List<U> users = bulkUsersDto.getUsers().stream()
+                .map(dto -> (U) conversionService.convert(dto, User.class))
+                .collect(Collectors.toList());
+        updateFields(users, tenants, emailConfirmationRequired, bulkUsersDto.getPassword());
+
+        String registrantToken = "";
+        String callbackUrl = "";
+        if (emailConfirmationRequired) {
+            CommonUserRegistrationDTO userDto = bulkUsersDto.getUsers().stream()
```

Comment:
Do I miss anything, or as soon as one user confirms his/her email, all other users will have their emails verified? (or it will crash) So the question is - why all users receive the same email?